### ISSUE TITLE
SlidingMenu Peek Margin enhancement

### ISSUE THREAD

**Issue by McPo** [jfeinstein10/SlidingMenu#65]
Provide the ability to offset the top layout, so has to provide a window to the bottom view. As oppose to completely covering the views all the time.

This is useful on tablets when using the XML version as oppose to the Root Activity version. A List including text and icon could be collapse to show only the icons. 

https://androiduiux.files.wordpress.com/2012/06/iconified_side_navigation.png

http://androiduiux.files.wordpress.com/2012/07/create-curiosity.png

**Comment by jfeinstein10** [comment]
Shouldn't the content (above portion) of the SlidingMenu be resized to fit its new view port? i.e. there should be some padding on the right? I see that the example screenshots have not done this

**Comment by McPo** [comment]
Yes it should. Sorry about that, I just grabbed the first screenshots I could from the web . (They arnt mine) the menu should be behind the content. And the content (top view ) should move all they way to the right, with a bit of padding Like the latest Google+ app. The menu (bottom view should be static)

Thanks for pointing that out.

**Comment by jfeinstein10** [comment]
Okay cool, that's what I thought; thanks for clearing it up. Another question: do you think the menu should be interactive? or that it should open up when you press it? I'm torn on this one

**Comment by McPo** [comment]
I am not sure if understand the question. I would expect the menu to be fully opened. I can then drag the content over the menu text. I could then drag the content away from the menu to expose the text again. The menu should still have focused (be interactive) when closed and opened, to allow for scrolling and button selection.

This allows the user to learn what each icon represents by examining the text. When the user becomes comfortable with the menu it will be possible to leave the menu collapsed, while navigating, improving space constraints and speed (one less slide/button hit to open the menu)

Thanks

**Comment by McPo** [comment]
I understand the dilemma now.

I would suggest working towards an  interactive model , at the start. It has much more use functionality wise.

The other interaction is only good for showing that there is a menu behind it.

I do believe it would be an excellent idea to eventually allow both interactions as they serve different purposes. One for navigation the other for indication.

**Comment by ghost** [comment]
I'm also very interested in this feature (see Issue 153). I'd suggest that the menu is working/clickable in both styles (collapsed and opened), because otherwise I see no reason to show it anyway.
The above content fills the screen to its right/left edge. So wenn the screen has a width of 800dp and the collapsed menu has a peek width of 100dp, the above content should be 700dp. As soon as the menu gets fully opened the above content slides away like it does now.
I think this is the same behavior as McPo proposed, isn't it?

**Comment by ghost** [comment]
Hi there!

Are there any news on this feature? Really looking forward to use this peek margin!

Thanks so far,
flo

**Comment by jzafrilla** [comment]
Hi! Anyone with a solution for this? i'm trying modifying CustomViewAbove class, but i can't get this feature...Thanks

**Comment by mjsibbald** [comment]
How do I accomplice this effect? Could anybody give me some pointers.

**Comment by Tadelaide** [comment]
come on 加油！although I don't know, but I do gain something about  slidemenu ☺️

**Comment by mjsibbald** [comment]
Not sure what you are saying here.

**Comment by McPo** [comment]
This library should be deprecated. Google has provided this functionality within a support library.
More info here http://developer.android.com/training/implementing-navigation/nav-drawer.html
